fix invalid escape sequences (python 3.12 SyntaxWarning)

ehossack opened the pull request:
I don't see any reason to not do this (nothing in sphinx?).

As referenced in https://github.com/Backblaze/b2-sdk-python/issues/458

ehossack commented:
(I could add a changelog section, but not sure what version this would target. 1.30.0? 1.29.1?

mjurbanski-reef commented on b2sdk/raw_simulator.py:
```diff
@@ -1159,7 +1159,7 @@ class RawSimulator(AbstractRawApi):
         DOWNLOAD_URL + '(?:' + '|'.join(
             (
-                '/b2api/v[0-9]+/b2_download_file_by_id\?fileId=(?P<file_id>[^/]+)',
+                '/b2api/v[0-9]+/b2_download_file_by_id?fileId=(?P<file_id>[^/]+)',
```
wow, this seems like this regexp was invalid from the beginning, right?

instead "fix" should be just adding `r` i.e. `r'...'` as otherwise this `?` will mean `d` is optional

ehossack replied:
good point.

mjurbanski-reef commented:
As for the changelog, we use towncrier to manage main file, so you just create a new one and don't worry about versions yourself.
Please instructions in  https://github.com/Backblaze/b2-sdk-python/blob/master/CONTRIBUTING.md#changelog .

P.S. thanks for contributing

ehossack commented:
> @ehossack the changelog files need to follow a specific naming scheme. ~If it's a fix, it should be `+<unique-change-name>.fixed.md`, e.g. `+syntax_warning.fixed.md`. Once you rename the file, it should be good to go. :)~
> 
> _Edit_: since this fixes issue #458, the name of the changelog file should be `458.fixed.md`.

Sorry, new process and also I didn't properly read the section you linked 😅

mjurbanski-reef reviewed (approved):
looks great, thank you!
